include/mvd/bits/sonata_misc.hpp

```diff
@@ -71,24 +73,68 @@ inline Positions SonataFile::getPositions(const Range & range) const{
 }
 
 inline Rotations SonataFile::getRotations(const Range & range) const{
-    Positions res{boost::extents[range.count > 0 ? range.count : size_][4]};
+    using Quaternion = boost::math::quaternion<double>;
 
-    auto xs = pop_->getAttribute<double>("orientation_x", select(range, size_));
-    auto ys = pop_->getAttribute<double>("orientation_y", select(range, size_));
-    auto zs = pop_->getAttribute<double>("orientation_z", select(range, size_));
-    auto ws = pop_->getAttribute<double>("orientation_w", select(range, size_));
+    Positions res{boost::extents[range.count > 0 ? range.count : size_][4]};
 
-    // No direct slicing write access from std::vector
-    for (size_t i = 0; i < (range.count > 0 ? range.count : size_); ++i) {
-        res[i][0] = xs[i];
-        res[i][1] = ys[i];
-        res[i][2] = zs[i];
-        res[i][3] = ws[i];
+    const auto attrs = pop_->attributeNames();
+    const bool quat = (attrs.count("orientation_x") +
+                       attrs.count("orientation_y") +
+                       attrs.count("orientation_z") +
+                       attrs.count("orientation_w")) == 4;
+
+    if (quat) {
+        auto xs = pop_->getAttribute<double>("orientation_x", select(range, size_));
+        auto ys = pop_->getAttribute<double>("orientation_y", select(range, size_));
+        auto zs = pop_->getAttribute<double>("orientation_z", select(range, size_));
+        auto ws = pop_->getAttribute<double>("orientation_w", select(range, size_));
+
+        // No direct slicing write access from std::vector
```

Comment:
Tried using boost::multi_array_ref to wrap the vector? 
Also, I saw some examples with array_view for assignment in https://theboostcpplibraries.com/boost.multiarray

Reply:
I think that will get pretty ugly. Not sure if it's worth it for code called once when loading.